### ISSUE TITLE
Word Boundary Navigation Fails for Non-ASCII Characters in Caret Movement

### ISSUE THREAD

**Issue by nivedano** [anthropics/claude-code#4456]
**Bug Description**
moving the caret with ctrl-arrow do not respect word boundaries for non-ASCII/non-English words, probably treating all as whitespace

**Environment Info**
- Platform: win32
- Terminal: vscode
- Version: 1.0.61
- Feedback ID: b5c1cea1-d37f-4ed2-aa4c-ff113c6fb215

**Comment by vanarok** [comment]
I have the same problem on Linux. It reproduces both in the vscode terminal inside Claude Code and in other terminals (I checked two external ones). All of them had bash.

**Comment by vanarok** [comment]
Still occurring. I have on macos vscode 1.106.3 terminal.
